MUST handle migration (#585)
quicwg/multipath: draft-ietf-quic-multipath.md

(see {{Section 21.1.1.1 of QUIC-TRANSPORT}}) and the congestion control
limits for this path are respected.

Though multipath offers clients with better options for use of different

> network paths, a server may receive a connection ID associated with a used path ID

Reviewer: ```suggestion
network paths, a server could receive a connection ID associated with a used path ID
```
Isn't this "endpoint"?

> on different 4-tuples, e.g., due to NAT rebinding. 

Author: ```suggestion
on different 4-tuples, e.g., due to NAT rebinding.
```

Therefore a server MUST handle migration
as specified in {{Section 9.3 of QUIC-TRANSPORT}}

> and MUST use a new connection ID for the same path ID
> for path validation.

Author: The point is that for migration you have to use the same path ID. We can remove the word "new" as it was intended to imply that you have to change, only that if you do chance is has to be from the same path ID.

```suggestion
and MUST use a connection ID for the same path ID
for path validation.
```

> for path validation.

Reviewer 2: OK with Mirja's suggestion.

> for path validation.

Reviewer: I don't think that works.  Because in RFC 9000, there is only one path ID (in effect) and the requirements are more complex than this captures.  The same path ID is already explicit elsewhere, so I think that this only risks creating confusion, with or without the word "new".

> for path validation.

Reviewer 2: @martinthomson path migration can happen on any valid path, not just path 0. If we extend the requirements of RFC 9000 to multipath, then the text + Mirja's suggestion makes sense.

> for path validation.

Reviewer: That's not my point.  The text is correct, but it is insufficient; that is, necessary but not sufficient.  RFC 9000 is quite particular about the rules for which connection ID to use in the event of migration (see above).  This text loses that nuance.

> for path validation.

Reviewer: > {{Section 9.3 of RFC9000}} defines how migration works when migration occurs, which - for multipath connections - is where packets that use connection IDs associated an existing path ID move to a new address tuple.

(see {{Section 21.1.1.1 of QUIC-TRANSPORT}}) and the congestion control
limits for this path are respected.

> Though multipath offers clients with better options for use of different
> network paths, an endpoint could receive a connection ID associated with a used path ID
> on different 4-tuples, e.g., due to NAT rebinding.
> Therefore a server MUST handle migration
> as specified in {{Section 9.3 of QUIC-TRANSPORT}}
> and MUST use a new connection ID for the same path ID
> for path validation.

Author: ```suggestion
An endpoint could receive a packet with a connection ID 
associated to an active path ID but the packet's 4-tuple does not match the 4-tuple
currently used with that path ID. This MUST be treated as path migration,
as specified in {{Section 9.3 of QUIC-TRANSPORT}}, with the constraint that
all connection IDs used during path migration MUST be
associated with the same path ID for the path being migrated.
```
